tests/ProductionConfigTests.js

```diff
@@ -24,28 +24,710 @@ https://github.com/GPII/universal/blob/master/LICENSE.txt
+                method: "PUT",
+                termMap: {
+                    gpiiKey: "{testCaseHolder}.options.gpiiKey"
+                }
+            }
+        }
+    }
+});
+
+// Flowmanager tests for:
+// /user/%gpiiKey/login and /user/%gpiiKey/logout (as defined in gpii.tests.development.testDefs),
+// /health,
+// /ready,
+// and using the "carla" snapset to test that it cannot be modified:
+// /access_token,
+// /%gpiiKey/settings/%device
+// /%gpiiKey/settings
+gpii.tests.productionConfigTesting.testDefs = fluid.transform(gpii.tests.development.testDefs, function (testDefIn) {
     var testDef = fluid.extend(true, {}, testDefIn, {
-        config: {
-            configName: "gpii.config.untrusted.development.json5",
-            configPath: "%gpii-universal/gpii/configs"
+        name: "Flow Manager production tests",
+        expect: 16,
+        config: gpii.tests.productionConfigTesting.config,
+        components: {
+            healthRequest: {
+                type: "kettle.test.request.http",
+                options: {
+                    port: "9082",
+                    hostname: "flowmanager",
+                    path: "/health",
+                    method: "GET",
+                    expectedStatusCode: 200,
+                    expectedPayload: {"isHealthy": true}
+                }
+            },
+            readyRequest: {
+                type: "kettle.test.request.http",
+                options: {
+                    port: "9082",
+                    hostname: "flowmanager",
+                    path: "/ready",
+                    method: "GET",
+                    expectedStatusCode: 200,
+                    expectedPayload: {"isReady": true}
+                }
+            },
+            accessTokenRequest: {
+                type: "kettle.test.request.http",
+                options: {
+                    port: "9082",
+                    hostname: "flowmanager",
+                    path: "/access_token",
+                    method: "POST",
+                    expectedStatusCode: 200
+                }
+            },
+            lifeCycleRequest: {
+                type: "kettle.test.request.http",
+                options: {
+                    port: "9082",
+                    hostname: "flowmanager",
+                    path: fluid.stringTemplate("/%gpiiKey/settings/%device", {
+                        gpiiKey: gpii.tests.productionConfigTesting.carlaKey,
+                        device: encodeURIComponent(JSON.stringify(
+                            gpii.tests.productionConfigTesting.device
+                        ))}
+                    ),
+                    headers: {
+                        "Authorization": "Bearer token" // set at test run
+                    },
+                    method: "GET",
+                    expectedStatusCode: 200
+                }
+            },
+            putSettingsRequestFailure: { // can't update snapset (readonly)
```

Comment:
This is a good test to be shared by both development and production config tests.

Reply:
I took a stab at moving this "update snapset" test to a common shared test definitions file, but it failed to run in the development configuration.  I'm almost certain it's because there are no snapsets in the database in that configuration.

Are the keys and prefs safes for the development tests located here:  `.../universal/build/tests/dbData`?  (Never mind the special case of the key-without-a-prefsSafe -- I know where that one is).  I tried adding a snapset key/safe there, just to see if that would solve the problem, but no joy.

I don't know how the database is set up in the case of the development configuration.  So, either I've put the snapset in the wrong place, or it's not being picked up by any initialization stage, or something else.

How is the database set up in the development configuration?

Reply:
The pouchDB set up for running dev config loads [these files](https://github.com/GPII/universal/blob/master/gpii/node_modules/testing/src/PouchTestCaseHolder.js#L45-L52).

You are right that there's no snapset key in this database. The prefs files in testData/preferences directory are converted to "user" keys and loaded. See line 50, 51 in PouchTestCaseHolder.js.

It might be worthwhile to sync up the usage of testData/preferences files for running tests in the dev and production configs. Do a experiment with changing line 50, 51 to load data from `%gpii-universal/build/dbData/snapset` and find out if we can change to failed tests to use GPII keys loaded from `%gpii-universal/build/tests/dbData/gpiiKeys.json` at line 48, 49, which are user keys solely for running dev tests. If we can make this change, the generation of `%gpii-universal/build/dbData/user` as a part of [this line](https://github.com/GPII/universal/blob/master/package.json#L80) can be removed.

Reply:
Thanks!  The pouch configuration you linked to shows that carla's key and prefs safe are both loaded, but they are 'user' preferences.  (Which makes me wonder why my access token request for 'carla' found nothing -- well, my problem to fix.)

More importantly, looking over that list of key/prefs json files, I wonder if all the ones that are actually 'snapsets' in production are needed as 'user' for the development tests.  I'll try your suggestion to see if that's necessary.  If it isn't, the problem of sharing snapsets for testing between production and development becomes simpler.  :crossed_fingers:

Reply:
An update, @cindyli:  I made the change to lines 50 and 51 to use 'snapset' instead of 'user' prefs safes and ran the node tests, the browser tests, and the productions tests.  I *think* I've run *all* of the tests.  There were no test failures.  That confirms we can share snapsets between production and development test configurations.  \o/

I also took a peek at how/if this will change in regards to @the-t-in-rtf's GPII-3619 pull request.  It looks like [that is already using the 'snapset' prefs safes](https://github.com/GPII/universal/pull/728/files#diff-444655ad8e7a4f3a9c67d93118d67912R17), and not the 'user' prefs safes.